packages/babel-plugin-ember-test-metadata/__tests__/utils-test.js

```diff
@@ -64,19 +69,82 @@ describe('Unit | utils | getNormalizedFilePath', () => {
       filename: path.join('this', 'is', 'not-an-embroider', 'path'),
     },
   };
+  const projectConfiguration = {
+    pkg: {
+      name: 'test-app',
+      'ember-addon': {},
+    },
+  };
 
   it('returns stripped file path as expected', () => {
-    expect(getNormalizedFilePath(fileOpts.embroiderBuildPath)).toBe(
-      path.join('tests', 'acceptance', 'my-test.js')
-    );
-    expect(getNormalizedFilePath(fileOpts.embroiderBuildPathTwoEmbroiderTokens)).toBe(
+    expect(getNormalizedFilePath(fileOpts.embroiderBuildPath, projectConfiguration)).toBe(
       path.join('tests', 'acceptance', 'my-test.js')
     );
+    expect(
+      getNormalizedFilePath(
+        fileOpts.embroiderBuildPathTwoEmbroiderPathSegments,
+        projectConfiguration
+      )
+    ).toBe(path.join('tests', 'acceptance', 'my-test.js'));
   });
 
   it('returns unmodified file path when path does not include "embroider" as a segment', () => {
-    expect(getNormalizedFilePath(fileOpts.normalFilePath)).toBe(
+    expect(getNormalizedFilePath(fileOpts.normalFilePath, projectConfiguration)).toBe(
       path.join('this', 'is', 'not-an-embroider', 'path')
     );
   });
 });
+
+describe('_getParsedClassicFilepath', () => {
+  const projectConfiguration = {
+    pkg: {
+      name: 'test-app',
+      'ember-addon': {},
+    },
+  };
+  const pathSegments = ['test-app', 'tests', 'acceptance', 'my-test.js'];
+  const monoRepoProjectConfiguration = {
+    pkg: {
+      name: `@parent-repo-name${path.sep}test-app`,
+      'ember-addon': {},
+    },
+  };
+  const monoRepoPathSegments = [
+    '@parent-repo-name',
+    'test-app',
+    'tests',
+    'acceptance',
+    'my-test.js',
+  ];
```

Comment:
Wondering what the benefit of these constants is vs just having the assertion arrangement happening right in the test. If you do it outside, you force the reader of the test to look around and read all the pieces of the test in order to understand what it's doing.

The way to think about tests are, they're usually write once (or few!) and read many. That means people use the tests as a reference to the expected behavior of the code. If you optimize for the write case, that can come as an expense to the read case.

Reply:
Thanks for the wisdom shared. I've inlined them and it is more readable now, what the specific test is doing.